service.go

```diff
@@ -154,7 +154,7 @@ func (s *service) createHTTPComponent() (Component, error) {
 		b.WithRoutesBuilder(s.routesBuilder)
 	}
 
-	if s.middlewares != nil && len(s.middlewares) > 0 {
+	if len(s.middlewares) > 0 {
 		b.WithMiddlewares(s.middlewares...)
```

Comment:
what will happen if we do not check for nil when we expand?

Reply:
Nothing bad. It's going to work as expected.
https://play.golang.org/p/JpgZmHwW_S_Z

But for the maps, this concern is valid: https://play.golang.org/p/iEJnYtZ4Y1j